### PR TITLE
Add support for WebVTT

### PR DESCRIPTION
**Pull request by joksas**
I use <https://www.iana.org/assignments/media-types/text/vtt> for magic numbers, as suggested [here](https://github.com/gabriel-vasile/mimetype/pull/222#pullrequestreview-836628591).

I have very little experience with file signatures, so maybe that's stupid but I've excluded the fifth and last byte sequences in the link because they end with `EOF`; or is there a way to represent this using the existing functions?

### CODE REVIEW

File: internal/magic/text.go [hunk 1]
```diff
@@ -120,6 +120,17 @@ var (
 	)
 	// Rtf matches a Rich Text Format file.
 	Rtf = prefix([]byte("{\\rtf1"))
+	// Vtt matches a Web Video Text Tracks (WebVTT) file. See
```
Comment (gabriel-vasile) [inline internal/magic/text.go:123]: I guess you wanted to add a link to IANA here. Feel free to do so.

**Review by gabriel-vasile — COMMENTED** [review]
I think you should add the checks for EOF too because IANA recommends so.
Unfortunately `prefix` cannot handle EOF so you'll have to write `Vtt` function yourself, similar to [zstd](https://github.com/gabriel-vasile/mimetype/blob/db2464c28aec4d6b0bf14bebba595b1ca4061d75/internal/magic/archive.go#L49) for example. You can use `bytes.HasPrefix` for non-EOF cases and `bytes.Equal` for EOF cases.

When it comes to MIME types and magic numbers IANA is THE authority, so I try to follow them as much as possible.

File: internal/magic/text.go (hunk 1 above)
Comment (joksas) [reply internal/magic/text.go]: That's exactly what it was. Thank you for noticing.

File: internal/magic/text.go [hunk 2]
```diff
@@ -348,3 +352,35 @@ func Srt(in []byte, _ uint32) bool {
...
+				continue
+			}
+			sig := append(sigStart, *sigEnd)
+			if bytes.HasPrefix(raw, sig) {
+				return true
+			}
+		}
+	}
+	return false
+}
```
Comment (gabriel-vasile) [inline internal/magic/text.go:386]: I think simple is better.  
```go
func Vtt(raw []byte, _ uint32) bool {
	prefixes := [][]byte{
		{the},
		{eight},
		{sequences},
		{without},
		{EOF},
	}

	for _, p := range prefixes {
		if bytes.HasPrefix(raw, p) {
			return true
		}
	}

	return bytes.Equal(raw, []byte{first EOF sequence}) ||
		bytes.Equal(raw, []byte{second EOF sequence})
}
```

**Review by gabriel-vasile — COMMENTED** [review]
Hi, sorry for the late reply. I left two suggestions.

File: internal/magic/text.go [hunk 3]
```diff
@@ -348,3 +352,35 @@ func Srt(in []byte, _ uint32) bool {
 	// A third line must exist and not be empty. This is the actual subtitle text.
 	return s.Scan() && len(s.Bytes()) != 0
 }
+
+// Vtt matches a Web Video Text Tracks (WebVTT) file. See
+// <https://www.iana.org/assignments/media-types/text/vtt>.
```
[the remark this answers is not in the document]
Comment (joksas) [reply internal/magic/text.go]: Resolved with f2d313bd4c524967711f421ef47cf52c6eeef71f.

File: internal/magic/text.go (hunk 2 above)
Comment (joksas) [reply internal/magic/text.go]: Resolved with af85bd637819ac4da4543e81cb5b920db876a489.

**Comment by joksas** [comment]
Hi,

Thank you for the comments. Should be fine now, but please let me know if I need to change anything else.

**Pull request merged**